source/common/formatter/substitution_formatter.cc

```diff
@@ -685,431 +685,430 @@ class StreamInfoSslConnectionInfoFieldExtractor : public StreamInfoFormatter::Fi
-               });
-         }},
-        {"UPSTREAM_HEADER_BYTES_SENT",
-         []() {
-           return std::make_unique<StreamInfoUInt64FieldExtractor>(
-               [](const StreamInfo::StreamInfo& stream_info) {
-                 return stream_info.getUpstreamBytesMeter()->headerBytesSent();
-               });
-         }},
-        {"DOWNSTREAM_WIRE_BYTES_SENT",
-         []() {
-           return std::make_unique<StreamInfoUInt64FieldExtractor>(
-               [](const StreamInfo::StreamInfo& stream_info) {
-                 return stream_info.getDownstreamBytesMeter()->wireBytesSent();
-               });
-         }},
-        {"DOWNSTREAM_HEADER_BYTES_SENT",
-         []() {
-           return std::make_unique<StreamInfoUInt64FieldExtractor>(
-               [](const StreamInfo::StreamInfo& stream_info) {
-                 return stream_info.getDownstreamBytesMeter()->headerBytesSent();
-               });
-         }},
-        {"DURATION",
-         []() {
-           return std::make_unique<StreamInfoDurationFieldExtractor>(
-               [](const StreamInfo::StreamInfo& stream_info) {
-                 return stream_info.requestComplete();
-               });
-         }},
-        {"RESPONSE_FLAGS",
-         []() {
-           return std::make_unique<StreamInfoStringFieldExtractor>(
-               [](const StreamInfo::StreamInfo& stream_info) {
-                 return StreamInfo::ResponseFlagUtils::toShortString(stream_info);
-               });
-         }},
-        {"UPSTREAM_HOST",
-         []() {
-           return StreamInfoAddressFieldExtractor::withPort(
-               [](const StreamInfo::StreamInfo& stream_info)
-                   -> std::shared_ptr<const Envoy::Network::Address::Instance> {
-                 if (stream_info.upstreamInfo() && stream_info.upstreamInfo()->upstreamHost()) {
-                   return stream_info.upstreamInfo()->upstreamHost()->address();
-                 }
-                 return nullptr;
-               });
-         }},
-        {"UPSTREAM_CLUSTER",
-         []() {
-           return std::make_unique<StreamInfoStringFieldExtractor>(
-               [](const StreamInfo::StreamInfo& stream_info) {
-                 std::string upstream_cluster_name;
-                 if (stream_info.upstreamClusterInfo().has_value() &&
-                     stream_info.upstreamClusterInfo().value() != nullptr) {
-                   if (Runtime::runtimeFeatureEnabled(
-                           "envoy.reloadable_features.use_observable_cluster_name")) {
-                     upstream_cluster_name =
-                         stream_info.upstreamClusterInfo().value()->observabilityName();
-                   } else {
-                     upstream_cluster_name = stream_info.upstreamClusterInfo().value()->name();
+const StreamInfoFormatter::FieldExtractorLookupTbl& StreamInfoFormatter::getKnownFieldExtractors() {
+  CONSTRUCT_ON_FIRST_USE(
```

Comment:
this looks a little strange -- is this constructing the hash-table and then copying it? Can you try:

```
  CONSTRUCT_ON_FIRST_USE(FieldExtractorLookupTbl, {
    {"REQUEST_DURATION",
    .....
   });

Reply:
Agree. There was probably extra copy there.